simapp/app.go

```diff
@@ -213,6 +216,36 @@ func NewSimApp(
 	tkeys := sdk.NewTransientStoreKeys(paramstypes.TStoreKey)
 	memKeys := sdk.NewMemoryStoreKeys(capabilitytypes.MemStoreKey)
 
+	// configure state listening capabilities using AppOptions
+	listeners := cast.ToStringSlice(appOpts.Get("store.streamers"))
+	for _, listenerName := range listeners {
+		// get the store keys allowed to be exposed for this streaming service/state listeners
+		exposeKeyStrs := cast.ToStringSlice(appOpts.Get(fmt.Sprintf("streamers.%s.keys", listenerName)))
+		exposeStoreKeys := make([]sdk.StoreKey, 0, len(exposeKeyStrs))
+		for _, keyStr := range exposeKeyStrs {
+			if storeKey, ok := keys[keyStr]; ok {
+				exposeStoreKeys = append(exposeStoreKeys, storeKey)
+			}
+		}
+		// get the constructor for this listener name
+		constructor, err := streaming.NewServiceConstructor(listenerName)
+		if err != nil {
+			tmos.Exit(err.Error()) // or continue?
+		}
+		// generate the streaming service using the constructor, appOptions, and the StoreKeys we want to expose
+		streamingService, err := constructor(appOpts, exposeStoreKeys, appCodec)
+		if err != nil {
+			tmos.Exit(err.Error())
+		}
+		// register the streaming service with the BaseApp
+		bApp.SetStreamingService(streamingService)
+		// waitgroup and quit channel for optional shutdown coordination of the streaming service
+		wg := new(sync.WaitGroup)
+		quitChan := make(chan struct{})
+		// kick off the background streaming service loop
+		streamingService.Stream(wg, quitChan)
+	}
```

Comment:
nit, can we move this to a public function somewhere? this constructor is getting too cluttered

Reply:
Yeah that's a good idea, will do!